src/player.js

```diff
@@ -636,7 +637,39 @@ export default class Player extends FakeEventTarget {
     this.dispatchEvent(new FakeEvent(CustomEventType.PLAYER_DESTROY));
     this._eventManager.destroy();
   }
+  /**
+   * Listen to the video element events and triggers them from the engine.
+   * @private
+   * @returns {void}
+   */
+  _attachMediaSource(): void {
+    if (this._engine) {
+      this._engine.attachMediaSource();
+    }
+  }
 
+  /**
+   * Listen to the video element events and triggers them from the engine.
+   * @private
+   * @returns {void}
+   */
+  _detachMediaSource(): void {
+    if (this._engine) {
+      this._engine.detachMediaSource();
+    }
+  }
+  /**
+   * handle ads with MSE.
+   * @private
+   * @returns {void}
+   */
+  _handleAdsWithMSE(): void {
```

Comment:
if we pass this call to `_attachMedia` you can basically just inline it in `_attachMedia` method.